lib/src/models/subreddit.dart

```diff
@@ -423,14 +420,80 @@ class Subreddit extends SubredditRef with RedditBaseInitializedMixin {
   }
 }
 
-// TODO(bkonyi): implement.
-// Provides functions to interact with the special [Subreddit]'s filters.
-/*class SubredditFilters {
-  final Subreddit _subreddit;
-  SubredditFilters(this._subreddit) {
-    throw new DRAWUnimplementedError();
+/// Provides functions to interact with the special [Subreddit]'s filters.
+class SubredditFilters {
+  static final RegExp _userRegExp = new RegExp(r'{user}');
+  static final RegExp _specialRegExp = new RegExp(r'{special}');
+  final SubredditRef _subreddit;
+
+  SubredditFilters._(this._subreddit) {
+    if ((_subreddit.displayName != 'all') &&
+        (_subreddit.displayName != 'mod')) {
+      throw new DRAWArgumentError(
+          'Only special Subreddits can be filtered (r/all or r/mod)');
+    }
+  }
+
+  /// Returns a stream of all filtered subreddits.
+  Stream<SubredditRef> call() async* {
+    final user = await _subreddit.reddit.user.me();
+    final path = apiPath['subreddit_filter_list']
+        .replaceAll(_specialRegExp, _subreddit.displayName)
+        .replaceAll(_userRegExp, user.displayName);
+    final Multireddit response = await _subreddit.reddit.get(path);
+    final filteredSubs = response.data['subreddits'];
+    for (final sub in filteredSubs) {
+      yield _subreddit.reddit.subreddit(sub['name']);
```

Comment:
This is purely for educational purposes:

I remember reading somewhere that functions are first class citizens, but since yield requires the function to have `async*`, how would you get the following to work/Is it even possible?

```
var filter = (subs) => yield _subreddit.reddit.subreddit(subs['name']);
filteredSubs.forEach(filter);
```

Reply:
You might be able to do something like this (I haven't tried it though):

```dart
var filter = (subs) async* => yield _subreddit.reddit.subreddit(subs['name']);
```